test/Microsoft.ML.CpuMath.UnitTests/UnitTests.cs

```diff
@@ -89,6 +90,20 @@ static CpuMathUtilsUnitTests()
 
             _testDstVectors = new AlignedArray[] { testDstVectorAligned1, testDstVectorAligned2 };
 
+            if (SkipAvxSse)
+            {
+                _disableAvxEnvironmentVariables = new Dictionary<string, string>()
+                {
+                    { _disableAvx , "0" }
+                };
+
+                _disableAvxAndSseEnvironmentVariables = new Dictionary<string, string>()
+                {
+                    { _disableAvx , "0" },
+                    { _disableSse , "0" }
+                };
+            }
+
 #if NETCOREAPP3_1
```

Comment:
Did you mean to remove this code? It looks duplicated with the code directly above it.

Reply:
It is the same code, but I wasn't sure exactly how to remove it. `SkipAvxSse` is set to true when running on arm/arm64 and is used elsewhere in the tests to skip some sections of the tests. This section you have highlighted here those values need to be set if 'SkipAvxSse' is set to be true, _or_ if we are targeting .netcore3.1, but they aren't mutually exclusive. I wasn't sure how to combine them so I left them as is.

Any ideas?

Reply:
Can we just have:

```C#
 if (SkipAvxSse || IsNetCoreApp3_1OrHigher())
{
....
}
```

?

Reply:
Something like this and then check if its higher?
```csharp
var framework = Assembly
    .GetEntryAssembly()?
    .GetCustomAttribute<TargetFrameworkAttribute>()?
    .FrameworkName;
```

Or is there an easier way?

Reply:
We do this in dotnet/runtime:

https://github.com/dotnet/runtime/blob/2be4fdc81f7f190ac35646ccc18799deb05da7dd/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs#L24

Reply:
So that returns interesting results. I made a sample project and this is what it returns.
```
PS C:\Repos\testing> dotnet run --framework net5.0
Environment.Version.Major: 5
RuntimeInformation.FrameworkDescription: .NET 5.0.7

PS C:\Repos\testing> dotnet run --framework netcoreapp3.1
Environment.Version.Major: 3
RuntimeInformation.FrameworkDescription: .NET Core 3.1.16

PS C:\Repos\testing> dotnet run --framework netcoreapp2.1
Environment.Version.Major: 4
RuntimeInformation.FrameworkDescription: .NET Core 4.6.30015.01
```
I'm trying to figure out a good way to exclude netcore 2 from this. Since all its versions are higher than the versions from netcore 3 though it does make it more annoying. Still working on exactly how to do this.

Reply:
Note that 2.1 goes EOL in a month. So I wouldn’t put too much effort into it. Maybe just hard code ‘4.6’ && “.NET Core” means “2.1”.

Reply:
Good point. I'll just hard code it to that then.

Reply:
Before .Net Core 3 according to https://docs.microsoft.com/en-us/dotnet/api/system.environment.version?view=net-5.0#remarks Version.Major should be 4 and Version.Minor should be 0. So I am now using that to check and the duplicate code has been removed.